ocaml/testsuite/tests/typing-layouts/annots.ml

```diff
@@ -79,6 +134,19 @@ Error: This alias is bound to type int list
 |}]
 (* CR layouts: error message could be phrased better *)
 
+let x : int list as ('a : value mod global) = [3;4;5]
+[%%expect {|
+Line 1, characters 8-43:
+1 | let x : int list as ('a : value mod global) = [3;4;5]
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: This alias is bound to type int list
+       but is used as an instance of type ('a : value mod global)
+       The layout of int list is value, because
+         it's a boxed variant type.
+       But the layout of int list must be a sublayout of value, because
```

Comment:
This clearly is a poor error message. I know doing better will be hard. And maybe best to do in a followup PR. In the meantime, I wonder how it would work to print a layout when all the modes are top, or print `immediate`/`immediate64` for those cases. Otherwise, print the layout `mod` all the non-top modes.

EDIT: Maybe that's already happening, but this error message just reports layout? Read my other comments before taking action here, as I suggest possibly moving error messages to talk about kinds, not layouts.